Iv 614

35359595 opened the pull request:
## Describe your changes
Tests for thea pallet including one for on_initialize and un-ignored one for BLS signature validation

## Issue ticket number and link
Closes #614 

## Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.
- [x] I removed all Clippy and Formatting Warnings. 
- [x] I added required Copyrights.

35359595 committed:
initial fixes

35359595 committed:
all tests up and passing

35359595 committed:
un-ignoring previously ignored test

35359595 committed:
on_initialize test

35359595 committed:
Merge branch 'Develop' of github.com:Polkadex-Substrate/Polkadex into iv-614

35359595 committed:
post-merge types added to mocks

serhii-temchenko reviewed (changes requested):
Not tested next cases:
1) - [x] `approve_deposit` with invalid origin
2) - [x]  Throw of `TokenTypeNotHandled` in `approve_deposit` extrinsic e.g.  by providing `TokenType` arg with `NonFungible`, `Generic` or `Fungible` witch value other than 1 or 2 - should return this error ([here](https://github.com/Polkadex-Substrate/Polkadex/blob/9d2add957b3195b1c259f66046ae99d426341da7/pallets/thea/src/lib.rs#L979-L990)).
3) - [x] `BoundedVectorOverflow` [here](https://github.com/Polkadex-Substrate/Polkadex/blob/9d2add957b3195b1c259f66046ae99d426341da7/pallets/thea/src/lib.rs#L907) is not tested.
4) - [x] [claim_deposit](https://github.com/Polkadex-Substrate/Polkadex/blob/9d2add957b3195b1c259f66046ae99d426341da7/pallets/thea/src/lib.rs#L404-L449) extrinsic is not tested at all. (in the list of related issue).
5) - [x] [batch_withdrawal_complete](https://github.com/Polkadex-Substrate/Polkadex/blob/9d2add957b3195b1c259f66046ae99d426341da7/pallets/thea/src/lib.rs#L451-L477) extrinsic is not tested at all. (in the list of related issue).
6) - [x] [withdraw](https://github.com/Polkadex-Substrate/Polkadex/blob/9d2add957b3195b1c259f66046ae99d426341da7/pallets/thea/src/lib.rs#L479-L503) extrinsic failing cases not tested like: 
- - [x] `BadOrigin` error [here](https://github.com/Polkadex-Substrate/Polkadex/blob/9d2add957b3195b1c259f66046ae99d426341da7/pallets/thea/src/lib.rs#L498). 
- - [x] `TheaKeyRotationInPlace` [here](https://github.com/Polkadex-Substrate/Polkadex/blob/9d2add957b3195b1c259f66046ae99d426341da7/pallets/thea/src/lib.rs#L740).
- - [x] WithdrawalNotAllowed [here](https://github.com/Polkadex-Substrate/Polkadex/blob/9d2add957b3195b1c259f66046ae99d426341da7/pallets/thea/src/lib.rs#L745).
7) - [x] `set_withdrawal_fee` BadOrigin error not tested [here](https://github.com/Polkadex-Substrate/Polkadex/blob/9d2add957b3195b1c259f66046ae99d426341da7/pallets/thea/src/lib.rs#L517).
8) - [x] [thea_key_rotation_complete](https://github.com/Polkadex-Substrate/Polkadex/blob/9d2add957b3195b1c259f66046ae99d426341da7/pallets/thea/src/lib.rs#L523-L580) is not tested at all.
9) - [x] [set_thea_key_complete](https://github.com/Polkadex-Substrate/Polkadex/blob/9d2add957b3195b1c259f66046ae99d426341da7/pallets/thea/src/lib.rs#L582-L630) is not tested at all.
10) - [x] [thea_queued_queued_public_key](https://github.com/Polkadex-Substrate/Polkadex/blob/9d2add957b3195b1c259f66046ae99d426341da7/pallets/thea/src/lib.rs#L632-L687) is not tested at all.
11) - [x] [thea_relayers_reset_rotation](https://github.com/Polkadex-Substrate/Polkadex/blob/9d2add957b3195b1c259f66046ae99d426341da7/pallets/thea/src/lib.rs#L689-L717) is not tested at  all.

35359595 committed:
deposit and router

35359595 committed:
claim_deposit test

35359595 committed:
batch_withdrawal_complete test

35359595 committed:
withdraw errors returning testing

35359595 committed:
test withdrawal fee origins

35359595 committed:
missing extrinsics tests

35359595 committed:
claim deposit test works

35359595 committed:
claim deposit test error cases cover

35359595 committed:
Merge branch 'iv-614' of github.com:Polkadex-Substrate/Polkadex into iv-614

35359595 committed:
thea benchmarks initial

35359595 committed:
claim_deposit benchmark

felixfaisal committed:
fix: batch withdrawal complete extrinsic

35359595 committed:
bls fixtures with sig verification failing

35359595 committed:
panic fix and queued relayers access

35359595 committed:
Merge branch 'iv-614' of github.com:Polkadex-Substrate/Polkadex into iv-614

felixfaisal committed:
fix: thea tests with signatures

35359595 committed:
Merge branch 'iv-614' into fx-614

35359595 committed:
fix: thea tests with signatures (#685)

35359595 committed:
all tests pass

35359595 committed:
fmt applied

35359595 commented:
> Not tested next cases:
> 
>     1. * [x]  `approve_deposit` with invalid origin
> 
>     2. * [x]   Throw of `TokenTypeNotHandled` in `approve_deposit` extrinsic e.g.  by providing `TokenType` arg with `NonFungible`, `Generic` or `Fungible` witch value other than 1 or 2 - should return this error ([here](https://github.com/Polkadex-Substrate/Polkadex/blob/9d2add957b3195b1c259f66046ae99d426341da7/pallets/thea/src/lib.rs#L979-L990)).
> 
>     3. * [x]  `BoundedVectorOverflow` [here](https://github.com/Polkadex-Substrate/Polkadex/blob/9d2add957b3195b1c259f66046ae99d426341da7/pallets/thea/src/lib.rs#L907) is not tested.
> 
>     4. * [x]  [claim_deposit](https://github.com/Polkadex-Substrate/Polkadex/blob/9d2add957b3195b1c259f66046ae99d426341da7/pallets/thea/src/lib.rs#L404-L449) extrinsic is not tested at all. (in the list of related issue).
> 
>     5. * [x]  [batch_withdrawal_complete](https://github.com/Polkadex-Substrate/Polkadex/blob/9d2add957b3195b1c259f66046ae99d426341da7/pallets/thea/src/lib.rs#L451-L477) extrinsic is not tested at all. (in the list of related issue).
> 
>     6. * [x]  [withdraw](https://github.com/Polkadex-Substrate/Polkadex/blob/9d2add957b3195b1c259f66046ae99d426341da7/pallets/thea/src/lib.rs#L479-L503) extrinsic failing cases not tested like:
> 
> 
>     * * [x]  `BadOrigin` error [here](https://github.com/Polkadex-Substrate/Polkadex/blob/9d2add957b3195b1c259f66046ae99d426341da7/pallets/thea/src/lib.rs#L498).
> 
>     * * [x]  `TheaKeyRotationInPlace` [here](https://github.com/Polkadex-Substrate/Polkadex/blob/9d2add957b3195b1c259f66046ae99d426341da7/pallets/thea/src/lib.rs#L740).
> 
>     * * [x]  WithdrawalNotAllowed [here](https://github.com/Polkadex-Substrate/Polkadex/blob/9d2add957b3195b1c259f66046ae99d426341da7/pallets/thea/src/lib.rs#L745).
> 
> 
>     7. * [x]  `set_withdrawal_fee` BadOrigin error not tested [here](https://github.com/Polkadex-Substrate/Polkadex/blob/9d2add957b3195b1c259f66046ae99d426341da7/pallets/thea/src/lib.rs#L517).
> 
>     8. * [x]  [thea_key_rotation_complete](https://github.com/Polkadex-Substrate/Polkadex/blob/9d2add957b3195b1c259f66046ae99d426341da7/pallets/thea/src/lib.rs#L523-L580) is not tested at all.
> 
>     9. * [x]  [set_thea_key_complete](https://github.com/Polkadex-Substrate/Polkadex/blob/9d2add957b3195b1c259f66046ae99d426341da7/pallets/thea/src/lib.rs#L582-L630) is not tested at all.
> 
>     10. * [x]  [thea_queued_queued_public_key](https://github.com/Polkadex-Substrate/Polkadex/blob/9d2add957b3195b1c259f66046ae99d426341da7/pallets/thea/src/lib.rs#L632-L687) is not tested at all.
> 
>     11. * [x]  [thea_relayers_reset_rotation](https://github.com/Polkadex-Substrate/Polkadex/blob/9d2add957b3195b1c259f66046ae99d426341da7/pallets/thea/src/lib.rs#L689-L717) is not tested at  all.

done

35359595 committed:
Merge branch 'iv-614'

serhii-temchenko reviewed (dismissed):
Nice result! Tests coverage increased on +12.30%!

felixfaisal commented on thea-primitives/src/lib.rs:
```diff
@@ -31,6 +31,10 @@ pub trait TheaExt {
 		let mut agg_pk: Option<AggregatePublicKey> = None;
 		for x in bit_map_vec {
+			// prevent panic on index out of bounds
+			if x as usize >= bls_public_keys.len() {
```
Let's say bls_public_key is empty, therefore it's length is `0` 
We get a map, whose 0th bit is set, Therefore it will be something like `for x in [0]` 

Then we have the if condition which will still work since `0` == `0` and will try to access the `0 index` element resulting in a panic.

35359595 replied:
good catch, I'm adding check for keys not been empty as start check in the method.

felixfaisal reviewed (dismissed):
Take a look at the scenario below, Let me know what you think @35359595

35359595 committed:
check if there are any keys prior sig verify

35359595 committed:
fmt applied

35359595 committed:
refactoring util + bench not compiling

35359595 committed:
dependencies cleanup

35359595 committed:
approve and cliem deposit extrinsics benchmarks

35359595 committed:
batch withdrawal complete benchmark

35359595 committed:
withdraw benchmark

35359595 committed:
set withdrawal fee benchmark

35359595 committed:
thea key rotation complete benchmark

35359595 committed:
thea key set complete benchmark

35359595 committed:
queued_queued and reset benchmarks

35359595 committed:
all benchmarks + deposit bug fix

35359595 committed:
benchmark machine generated weights

35359595 committed:
benchmarks generated weigts attached

35359595 committed:
test weight config in mock for thea

Gauthamastro committed:
Iv 606 (#689)

## Describe your changes

benchmarks and weights generated on weight server.  

## Issue ticket number and link

Closes #606 

## Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.
- [x] I removed all Clippy and Formatting Warnings. 
- [x] I added required Copyrights.

Gauthamastro committed:
Merge branch 'Develop' into iv-614

# Conflicts:
#	pallets/thea/src/lib.rs

Gauthamastro committed:
clippy fix

Gauthamastro commented:
@35359595 test's are failing

35359595 committed:
Merge branch 'Develop' into iv-614

35359595 committed:
missing imports

35359595 committed:
mock fixes

zktony committed:
Fixed tests

35359595 committed:
Merge branch 'Develop' of github.com:Polkadex-Substrate/Polkadex into iv-614

serhii-temchenko committed:
Removed unused dependency

serhii-temchenko commented:
> @35359595 test's are failing

Fixed. @Gauthamastro please review.
